src/main/java/io/javalin/plugin/openapi/dsl/extractDocumentation.kt

```diff
@@ -56,44 +55,59 @@ private fun HandlerMetaInfo.extractDocumentationWithPathScanning(options: Create
     }
 }
 
-private fun HandlerMetaInfo.getOpenApiAnnotationFromReference(): OpenApi? {
+private fun HandlerMetaInfo.getOpenApiAnnotationFromReference(): List<OpenApi> {
     return try {
-        methodReferenceOfHandler?.getOpenApiAnnotation()
-                ?: handler.lambdaField?.getOpenApiAnnotation()
+        methodReferenceOfHandler.mapNotNull(Method::getOpenApiAnnotation).ifEmpty {
+            handler.lambdaField?.getOpenApiAnnotation()?.let { listOf(it) } ?: listOf()
+        }
     } catch (e: NoSuchFieldException) {
-        null
+        listOf()
     } catch (e: Error) {
         return if (e::class.qualifiedName == "kotlin.reflect.jvm.internal.KotlinReflectionInternalError") {
             Logger.getGlobal()
                     .warning("Local functions, lambdas, anonymous functions and local variables with @OpenApi annotations are currently not supported. " +
                             "The annotation of the handler for \"$httpMethod $path\" will be ignored. To fix this, move the handler into a global function.")
-            null
+            listOf()
         } else {
             throw e
         }
     }
 }
 
-private fun HandlerMetaInfo.getOpenApiAnnotationFromHandler(): OpenApi? {
+private fun HandlerMetaInfo.getOpenApiAnnotationFromHandler(): List<OpenApi> {
     return try {
         val method = handler::class.java.getMethodByName("handle")!!
-        method.getOpenApiAnnotation()
+        method.getOpenApiAnnotation()?.let { listOf(it) } ?: listOf()
     } catch (e: NullPointerException) {
-        null
+        listOf()
     }
 }
 
-private val HandlerMetaInfo.methodReferenceOfHandler: Method?
+private val HandlerMetaInfo.methodReferenceOfHandler: List<Method>
     get() = when {
-        handler.isClass -> (handler as Class<*>).methods[0]
+        handler.isClass -> listOf((handler as Class<*>).methods[0])
         handler.isKotlinMethodReference -> {
-            val functionValue = handler.getFieldValue("function") as KFunction<*>
-            functionValue.javaMethod
+            val functionValue = handler.getFieldValue("function") as FunctionReference
+            val javaMethod = functionValue.javaMethod
+            if (javaMethod != null && functionValue.isAbstract) {
```

Comment:
Could you create some helper methods for the things inside the `if (javaMethod != null && functionValue.isAbstract)`

Reply:
sure